searching/exponential_search.c

```diff
@@ -0,0 +1,99 @@
+/**
+ *  \file
+ *  \@brief [Exponential Search](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/en/Search%20Algorithms/Exponential%20Search.md)
+ *  \author [Alessio Farinelli](https://github.com/faridevnz)
+ */
+#include <assert.h>  /// for assert
+
+#define ELEMENT -10
+
+int binary_search(const int* arr, const int l_index, const int r_index, const int n);
```

Comment:
Consider using `uint64_t`, `uint32_t`, `uint16_t`, or even `uint8_t` for non-negative values. For negative values, use `int64_t` or their appropriate size ([reference](https://en.cppreference.com/w/c/types/integer)). Requires the `inttypes.h` header. Check out other parts of the code, too. 🙂